project-icp/bee-pollinator-app
# Upgrade development environment to PostgreSQL 9.6

## Overview

In addition to adding support for PostgreSQL 9.6, upgrade PostGIS to 2.3, which coincides with the version utilized by RDS in staging and production.

Use the `ansible_local` provisioner to execute Ansible directly on the guest machine to avoid incompatibility with Ansible 2.9+ on hosts.

Lastly, the project specific init script for PostgreSQL was removed because it only existed to emit a `postgresql-started` event for the `pgweb` service, which is also being removed in this change set.

Resolves #527 

---

**Testing**

- Destroy any existing development environment virtual machines

```bash
$ vagrant destroy -f
```

- Bring up the new development environment; ensure it provisions cleanly

```bash
$ vagrant up --provision
```

- Execute migrations

```bash
$ ./scripts/manage.sh migrate
```

- Execute the Django test suite and ensure it completes successfully

```bash
$ ./scripts/manage.sh test
```

- Ensure health check is passing

```bash
$ http localhost:8000/health-check/
HTTP/1.1 200 OK
Connection: keep-alive
Content-Encoding: gzip
Content-Type: application/json
Date: Tue, 07 Jan 2020 17:23:24 GMT
Server: nginx
Transfer-Encoding: chunked
Vary: Accept-Encoding

{
    "caches": [
        {
            "default": {
                "ok": true
            }
        }
    ],
    "databases": [
        {
            "default": {
                "ok": true
            }
        }
    ]
}
```


- Lastly, validate the versions of PostgreSQL and PostGIS

```bash
$ vagrant ssh services
vagrant@services:~$ sudo su postgres
postgres@services:/home/vagrant$ psql
psql (9.6.13)
Type "help" for help.

postgres=# \l
                                  List of databases
   Name    |  Owner   | Encoding |   Collate   |    Ctype    |   Access privileges
-----------+----------+----------+-------------+-------------+-----------------------
 icp       | icp      | UTF8     | en_US.UTF-8 | en_US.UTF-8 |
 postgres  | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 |
 template0 | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 | =c/postgres          +
           |          |          |             |             | postgres=CTc/postgres
 template1 | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 | =c/postgres          +
           |          |          |             |             | postgres=CTc/postgres
(4 rows)

postgres=# \c icp
You are now connected to database "icp" as user "postgres".
icp=# SELECT version();
                                                                version
---------------------------------------------------------------------------------------------------------------------------------------
 PostgreSQL 9.6.13 on x86_64-pc-linux-gnu (Ubuntu 9.6.13-1.pgdg14.04+1), compiled by gcc (Ubuntu 4.8.4-2ubuntu1~14.04.4) 4.8.4, 64-bit
(1 row)

icp=# SELECT postgis_version();
            postgis_version
---------------------------------------
 2.3 USE_GEOS=1 USE_PROJ=1 USE_STATS=1
(1 row)
```
